### PR TITLE
Editor: Move the sidebar component to the editor package

### PR DESCRIPTION
**Pull request by youknowriad**
Related #52632

## What?

We're in a middle of an effort to merge the post and site editors under the @wordpress/editor abstraction. As part of this,  this PR moves the Sidebar component from the edit-post to the editor package.

The idea is that on a follow-up PR, I'll be removing the sidebar that is implemented in the edit-site package and just replace it with the one that is on the editor package (for both editors). I preferred to do this in two steps to avoid a huge PR. 

## Testing Instructions

Please test the sidebar (block and document) of the post editor. There should be no change compared to trunk.

### CODE REVIEW

File: packages/editor/src/components/sidebar/constants.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const sidebars = {
+	document: 'edit-post/document',
+	block: 'edit-post/block',
```
Comment (Mamaduka) [inline packages/editor/src/components/sidebar/constants.js:3]: Are we keeping the old namespace for now? Let's note that we should update these when sidebar-related store actions are also unified.

Comment (youknowriad) [reply packages/editor/src/components/sidebar/constants.js:3]: We can't change these for BC reasons unfortunately. We discussed this with @fabiankaegy in a previous PR and this was the best tradeoff

**Review by Mamaduka — APPROVED** [review]
I couldn't spot any regression.

The editor sidebar has good e2e test coverage, so it should be a good merge after all CI checks are green.

**Review by jameskoster — APPROVED** [review]
I couldn't find any regressions.

**Comment by sybrew** [comment]
The classnames changed; plugins that used to target the sidebar with their CSS have now regressed.

**Comment by youknowriad** [comment]
Hey @sybrew, sorry for the breakage.

Classnames and dom elements are not considered public APIs per our guidelines https://developer.wordpress.org/block-editor/contributors/code/backward-compatibility/#class-names-and-dom-updates That said, if we can ensure things don't break as we iterate on the UI, we're happy to do so. Can you share which classnames are you thinking about? Do you mind opening an issue with all the details.

**Comment by sybrew** [comment]
It's OK. I've detailed it here: https://github.com/sybrew/the-seo-framework/commit/5ee3395b2e73bc84fb612fc0dd6da08c8ffe4375.

We are continuing to rely on the meta box API until I find an opportunity to improve upon that, so I am not directly interacting with React.

The meta box implementation is still quite ugly, so now that I got your attention, please [consider implementing this UI fix](https://github.com/WordPress/gutenberg/issues/7960#issuecomment-1382336083).

**Pull request merged**